### ISSUE TITLE
after new setup at sign in "server appears to be unavailable"

### ISSUE THREAD

**Issue by gpciceri** [germinateplatform/gatekeeper-server#1]
Hi all,
I've followd the "docker_compose" instructions, it seems that gatekeeper and mysql are up and running, but when I sign in with "Admin/password" I take "Server appears to be unavailable". 

It obviously seems a problem of configuration (it happens even if I input awrong password, to try if something change). the "docker_compose up" command tells me that both gatekeeper and mysql are up and running.

So my question: where may I check for more detailed logging information? Any clue?

Thanks in advance, best regards
/gp

**Comment by sebastian-raubach** [comment]
That error is shown if an error code is returned from the server that isn't a 403 or a 400, so my guess would be it's a 500.
You can check the logs by starting an interactive docker session while docker-compose is up:

```
docker exec -it {GATEKEEPER_CONTAINER_ID} bash

cd /usr/local/tomcat/logs

ls -ltr
```

Then identify the latest `ROOT-{date}.log` file and check if there are any errors in there.

Is this the first time you've started everything together with docker-compose? If so, down it and up it again to see if that fixes it.

**Comment by gpciceri** [comment]
Hi,
thanks for the infos,
I've just down/up-ed the dockers and retested gatekeeper login: same situation.

I've checked our the ROOT tomcat log, it shows a chain of java exceptions - no way for me to understand what happened. I've attached it here - hope you can find it something useful.

[ROOT.2020-11-12.log](https://github.com/germinateplatform/gatekeeper-server/files/5528718/ROOT.2020-11-12.log)

If you'll tell me some debugging switch you think useful to run tomcat with, I'll re-try to have more infos, of course.

Thnaks a lot for your patience, have a nice day.
/gp


.

**Comment by sebastian-raubach** [comment]
Hi,

Would you mind sharing your docker-compose.yml file and the config.properties file of Gatekeeper here? Please replace usernames and passwords with "username" and "password", assuming they are identical between the two files.
Also, is there any other log file that contains anything that looks like a Gatekeeper related error, maybe in the `catalina-{date}.log` or the `localhost-{date}.log`?

Thanks

**Comment by gpciceri** [comment]
thanks fot the reply,

yes, sure: here they are (some file extensions rename was necessary, sorry):

[docker-compose_gatekeeper-20201112-yml.txt](https://github.com/germinateplatform/gatekeeper-server/files/5529246/docker-compose_gatekeeper-20201112-yml.txt)

[config_gatekeeper-20201112-properties.txt](https://github.com/germinateplatform/gatekeeper-server/files/5529254/config_gatekeeper-20201112-properties.txt)

and the logs (no infos I can find useful, but perhaps...)

[catalina.2020-11-12.log](https://github.com/germinateplatform/gatekeeper-server/files/5529260/catalina.2020-11-12.log)
[localhost_access_log.2020-11-12.txt](https://github.com/germinateplatform/gatekeeper-server/files/5529262/localhost_access_log.2020-11-12.txt)
[ROOT.2020-11-12.log](https://github.com/germinateplatform/gatekeeper-server/files/5529263/ROOT.2020-11-12.log)

As a consideration, it should be useful having an additional logging INFO message when tomcat, after its own startup, is able to connect successfully to mysql: I suppose I've misconfigurated this part of communication between tomcat and mysql (or perhaps this message already exists but I'm not able to find it, sorry again).

Any help will be highly appreciated, as usual.

Thanks a lot.
/gp

**Comment by sebastian-raubach** [comment]
Ok, I think there's an issue with the `database.server` property in your config.properties.
The way Docker works and specifically this Docker compose setup is, that both containers run on the same network. The advantage of that is, that they can talk to each other. This does mean, however, that Gatekeeper needs to use the docker-compose container name of MySQL for the database connection property.
In your case, that is `mysql_gatek`.
So setting `database.server` to `mysql_gatek` should allow Gatekeeper to connect to the database.

**Comment by gpciceri** [comment]
Thanks a lot,
it seems to be sligthy better (along with mysql_gatek hostname I've leaved
the default 3306 mysqld port), now:

0. if I type the admin password wrong I take "Invalid username or
password",
so the mysql connection should be OK (the user is stored in mysql, if I
understand);

1. BUT if I enter correctly "Admin/password" it simply hangs - not showing
the after-successful-login page (perhaps it will be a password-change
page), so I suppose there could be an issue with web.base parameter...

In the meanwhile, I've upgraded gatekeeper to 4.1.1, same behaviour.

Have you please some other thing to check to suggest to me?

Thanks in advance, best regards.
/gp



"Antiochus, when he was Ephor, hearing that Philip had given Messenians
their land, asked if he had also provided them with the power to prevail in
fighting to keep it." (Plutarch, Sayings of Spartans)


gian paolo ciceri
mobile : +39 347 9985882, skype: gpciceri


Il giorno gio 12 nov 2020 alle ore 09:55 Sebastian Raubach <
notifications@github.com> ha scritto:

**Comment by sebastian-raubach** [comment]
Can you check the network tab of your browser (usually found under F12) and see what the `/token` call returns? Also, is there anything in the logs after the login with correct credentials when it "hangs"?

**Comment by gpciceri** [comment]
Hurrah,
the /token was hanging, so I rechecked the config.

It was the salt parameter (I had changed it to 30), I've restored it to the
default 10 value. Now I can see the interface and I was able to change
Admin's password.

I close this issue - thanks a lot for your support.


/gp







"Antiochus, when he was Ephor, hearing that Philip had given Messenians
their land, asked if he had also provided them with the power to prevail in
fighting to keep it." (Plutarch, Sayings of Spartans)


gian paolo ciceri
mobile : +39 347 9985882, skype: gpciceri


Il giorno gio 12 nov 2020 alle ore 15:07 Sebastian Raubach <
notifications@github.com> ha scritto:
